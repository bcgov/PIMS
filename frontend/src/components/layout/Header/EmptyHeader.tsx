import './Header.scss';

import BClogoUrl from 'assets/images/logo-banner.svg';
import PIMSlogo from 'assets/images/PIMSlogo/logo_only.png';
import React from 'react';
import { Nav, Navbar } from 'react-bootstrap';
import styled from 'styled-components';

const VerticalBar = styled.span`
  border-left: 2px solid white;
  font-size: 34px;
  margin: 0 15px 0 25px;
  vertical-align: top;
`;

/**
 * Display an "empty" header bar with limited functionality as a placeholder
 */
const EmptyHeader = () => {
  return (
    <Navbar expand className="App-header">
      <Navbar.Brand className="brand-box">
        <a target="_blank" rel="noopener noreferrer" href="https://www2.gov.bc.ca/gov/content/home">
          <img
            className="bc-gov-icon"
            src={BClogoUrl}
            width="156"
            height="43"
            alt="Government of BC logo"
          />
        </a>
        <VerticalBar />
        <img className="pims-logo" src={PIMSlogo} height="50" alt="PIMS logo" />
      </Navbar.Brand>
      <Nav className="title mr-auto">
        <Nav.Item>
          <h1 className="longAppName">Property Inventory Management System</h1>
          <h1 className="shortAppName">PIMS</h1>
        </Nav.Item>
      </Nav>
    </Navbar>
  );
};

export default EmptyHeader;
