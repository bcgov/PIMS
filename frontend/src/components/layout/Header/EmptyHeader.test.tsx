import { render } from '@testing-library/react';
import React from 'react';

import EmptyHeader from './EmptyHeader';

describe('Empty Header', () => {
  it('renders', () => {
    const { container } = render(<EmptyHeader></EmptyHeader>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
