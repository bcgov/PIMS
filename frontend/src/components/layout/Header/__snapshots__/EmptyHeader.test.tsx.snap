// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Header renders 1`] = `
.c0 {
  border-left: 2px solid white;
  font-size: 34px;
  margin: 0 15px 0 25px;
  vertical-align: top;
}

<nav
  class="App-header navbar navbar-expand navbar-light"
>
  <span
    class="brand-box navbar-brand"
  >
    <a
      href="https://www2.gov.bc.ca/gov/content/home"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Government of BC logo"
        class="bc-gov-icon"
        height="43"
        src="() => /*#__PURE__*/(0, _jsxDevRuntime.jsxDEV)(\\"template\\", {
  children: /*#__PURE__*/(0, _jsxDevRuntime.jsxDEV)(\\"svg\\", {}, void 0, false, {
    fileName: _jsxFileName,
    lineNumber: 5,
    columnNumber: 5
  }, void 0)
}, void 0, false, {
  fileName: _jsxFileName,
  lineNumber: 4,
  columnNumber: 3
}, void 0)"
        width="156"
      />
    </a>
    <span
      class="c0"
    />
    <img
      alt="PIMS logo"
      class="pims-logo"
      height="50"
      src="logo_only.png"
    />
  </span>
  <div
    class="title mr-auto navbar-nav"
  >
    <div
      class="nav-item"
    >
      <h1
        class="longAppName"
      >
        Property Inventory Management System
      </h1>
      <h1
        class="shortAppName"
      >
        PIMS
      </h1>
    </div>
  </div>
</nav>
`;
